tools/pyboard.py

```diff
@@ -153,13 +153,72 @@ def inWaiting(self):
         else:
             return n_waiting
 
+# A thread that constantly reads from subp stdout and puts data into a queue (FIFO list)
+# When tested, it reads double '\r' which messes up the further code, so if double '\r' detected, skip one
```

Comment:
Do you understand why it reads double \r?

Reply:
I have tried to investigate the reason, but unfortunately I am still not able to understand why it reads double \r